src/shared_modules/rsync/src/rsyncImplementation.cpp

```diff
@@ -0,0 +1,321 @@
+    queryParam["column_list"] = querySelect.at("column_list");
+    queryParam["distinct_opt"] = querySelect.at("distinct_opt");
+    queryParam["order_by_opt"] = querySelect.at("order_by_opt");
+    
+    const std::unique_ptr<cJSON, CJsonDeleter> spJson{ cJSON_Parse(selectData.dump().c_str()) };
+    spDBSyncWrapper->select(spJson.get(), { callbackDBSync, &sizeRange });
+
+    return size;
+}
+
+
+void RSyncImplementation::fillChecksum(const std::shared_ptr<DBSyncWrapper>& spDBSyncWrapper, 
+                                             const nlohmann::json& jsonSyncConfiguration,
+                                             const std::string& begin,
+                                             const std::string& end,
+                                             ChecksumContext& ctx) 
+{
+    nlohmann::json selectData;
+    const auto& indexFieldName { jsonSyncConfiguration.at("index").get_ref<const std::string&>() };
+    selectData["table"] = jsonSyncConfiguration.at("table");
+    
+    const auto& querySelect { jsonSyncConfiguration.at("range_checksum_query_json") };
+    const auto& checksumFieldName { jsonSyncConfiguration.at("checksum_field").get_ref<const std::string&>() };
+    
+    const auto middle { ctx.size / 2 };
+    auto index { 1ull };
+
+    std::unique_ptr<Utils::HashData> hash{ std::make_unique<Utils::HashData>(Utils::HashType::Sha256) };
+    std::function<void(const nlohmann::json&)> calcChecksum
+    {
+        [&] (const nlohmann::json& resultJSON)
+        {
+            const auto checksumValue { resultJSON.at(0).at(checksumFieldName).get_ref<const std::string&>() };
+            hash->update(checksumValue.data(), checksumValue.size());
+
+            if (CHECKSUM_SPLIT == ctx.type)
+            {
+                if (middle+1 == index)
+                {
+                    ctx.rightCtx.begin = resultJSON.at(0).at(indexFieldName);
+                    ctx.leftCtx.tail = ctx.rightCtx.begin;
+                } 
+                else if(middle == index)
+                {
+                    ctx.leftCtx.end = resultJSON.at(0).at(indexFieldName);
+                    ctx.leftCtx.checksum = Utils::asciiToHex(hash->hash());
+                    hash = std::make_unique<Utils::HashData>(Utils::HashType::Sha256);
+                }
+
+                if(ctx.size == index)
+                {
+                    ctx.rightCtx.checksum = Utils::asciiToHex(hash->hash());
+                }
+            }
+            else
+            {
+                if(ctx.size == index)
+                {
+                    ctx.leftCtx.checksum = Utils::asciiToHex(hash->hash());
+                }
+            }
+            
+            ++index;
+        }
+    };
+
+    auto rowFilter { querySelect.at("row_filter").get_ref<const std::string&>() } ;
+    Utils::replaceFirst(rowFilter, "?", begin);
+    Utils::replaceFirst(rowFilter, "?", end);
+
+    auto& queryParam { selectData["query"] };
```

Comment:
selectData.at("query")

Reply:
.at return a const and is only for read, I need append into queryParam in the future.